hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/driver/impl/StateStoreFileBaseImpl.java

```diff
@@ -88,6 +88,15 @@ protected abstract <T extends BaseRecord> BufferedReader getReader(
   protected abstract <T extends BaseRecord> BufferedWriter getWriter(
       String path);
 
+  /**
```

Comment:
I think that separating this out from getWriter is confusing and unnecessary. Just make getWriter public in all 3 classes.

Reply:
Although do add the VisibleForTesting annotation.